File: src/main/java/org/redrune/networking/packet/context/impl/CommandPacketContext.java
package org.redrune.networking.packet.context.impl;

import org.redrune.game.content.plugin.PluginRepository;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.networking.packet.context.PacketContext;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
public class CommandPacketContext extends PacketContext {
	
	private final boolean clientCommand;
	private final boolean unknown;
	private final String command;
	
	public CommandPacketContext(boolean clientCommand, boolean unknown, String command) {
		this.clientCommand = clientCommand;
		this.unknown = unknown;
		this.command = command;
	}
	
	@Override
	public void handle(Player player) {
		PluginRepository.handleCommand(player, command.replaceFirst("::", "").split(" "), true, clientCommand);
	}
}

File: src/main/kotlin/org/redrune/networking/packet/context/impl/CommandPacketContext.kt
package org.redrune.networking.packet.context.impl

import org.redrune.game.content.plugin.PluginRepository.handleCommand
import org.redrune.game.entity.actor.player.Player
import org.redrune.networking.packet.context.PacketContext
import java.util.*

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
class CommandPacketContext(
    private val clientCommand: Boolean,
    private val unknown: Boolean,
    private val command: String,
) : PacketContext() {
    override fun handle(player: Player) {
        handleCommand(
            player,
            command.lowercase(Locale.getDefault()).replaceFirst("::".toRegex(), "").split(" ").toTypedArray(),
            true,
            clientCommand
        )
    }
}